File: Maximum Number of Balloons - Leetcode 1189/Maximum Number of Balloons - Leetcode 1189.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int maxNumberOfBalloons(String text) {
        Map<Character, Integer> counter = new HashMap<>();
        String balloon = "balloon";

        for (char c : text.toCharArray()) {
            if (balloon.indexOf(c) != -1) {
                counter.put(c, counter.getOrDefault(c, 0) + 1);
            }
        }

        if (!counter.containsKey('b') || 
            !counter.containsKey('a') || 
            !counter.containsKey('l') || 
            !counter.containsKey('o') || 
            !counter.containsKey('n')) {
            return 0;
        } else {
            return Math.min(Math.min(counter.get('b'), counter.get('a')), 
                            Math.min(counter.get('l') / 2, 
                                     Math.min(counter.get('o') / 2, counter.get('n'))));
        }
    }
}

File: Maximum Number of Balloons - Leetcode 1189/Maximum Number of Balloons - Leetcode 1189.js
var maxNumberOfBalloons = function(text) {
    const counter = {};
    const balloon = "balloon";

    for (const c of text) {
        if (balloon.includes(c)) {
            counter[c] = (counter[c] || 0) + 1;
        }
    }

    if (!('b' in counter) || 
        !('a' in counter) || 
        !('l' in counter) || 
        !('o' in counter) || 
        !('n' in counter)) {
        return 0;
    } else {
        return Math.min(counter['b'], counter['a'], Math.floor(counter['l'] / 2), Math.floor(counter['o'] / 2), counter['n']);
    }
};
